src/components/Navbar/index.tsx

```diff
@@ -5,32 +5,56 @@ import {
   useAuthStateContext
 } from "../../context/auth";
 import { Select } from "antd";
+import { useQuery } from "@apollo/react-hooks";
+import {
+  GET_PROJECTS,
+  ProjectsQuery,
+  Project
+} from "graphql/queries/get-projects";
 
 const { Option, OptGroup } = Select;
 
+const renderProjectOption = (isFavorite: boolean = false) => ({
+  identifier,
+  displayName
+}: Project) => (
+  // two Options cannot have the same value attribute or it breaks ability to scroll with keyboard.
+  // therefore "-favorite" is appended to the end of a favorite's value prop
+  <Option
+    key={identifier}
+    value={isFavorite ? `${identifier}-favorite` : identifier}
+  >
+    {displayName}
+  </Option>
+);
+
 export const Navbar: React.FC = () => {
   const { logout } = useAuthDispatchContext();
   const { isAuthenticated } = useAuthStateContext();
 
+  const { data, loading } = useQuery<ProjectsQuery>(GET_PROJECTS);
+
   return (
     <Wrapper>
       <InnerWrapper>
         <StyledSelect
           showSearch={true}
-          placeholder="Branch"
+          placeholder="Project"
           optionFilterProp="children"
-          defaultValue={["spruce"]}
+          loading={loading}
+          disabled={loading}
         >
-          <OptGroup label="Favorites">
-            <Option value="evergreen-favorite">Evergreen</Option>
-            <Option value="spruce-favorite">Spruce</Option>
-          </OptGroup>
-          <OptGroup label="Engineer">
-            <Option value="evergreen">Evergreen</Option>
-            <Option value="spruce">Spruce</Option>
-            <Option value="cloud">Cloud</Option>
-            <Option value="build">Build</Option>
-          </OptGroup>
+          {data && data.projects.favorites.length > 0 && (
+            <OptGroup label="Favorites">
+              {data.projects.favorites.map(renderProjectOption(true))}
+            </OptGroup>
+          )}
+          {data &&
```

Comment:
I believe this will duplicate the favorite projects won't it? I think having them in there twice would be more confusing than just sorting favorites to the top

Reply:
Yes, projects that are present in favorites are still present in in the rest of the dropdown. This seems like the right UX to me because favoriting something does not mean removing it from the group to which it belongs. But I agree filtering favorites out of the projects would make this feature simpler and prevent problems caused by duplication.